language/tools/move-mv-llvm-compiler/src/stackless/llvm.rs

```diff
@@ -101,6 +103,33 @@ impl Context {
     pub fn int256_type(&self) -> Type {
         unsafe { Type(LLVMIntTypeInContext(self.0, 256)) }
     }
+    pub fn int_arbitrary_type(&self, len: usize) -> Type {
+        unsafe { Type(LLVMIntTypeInContext(self.0, len as libc::c_uint)) }
+    }
+
+    pub fn array_type(&self, ll_elt_ty: Type, len: usize) -> Type {
+        unsafe { Type(LLVMArrayType(ll_elt_ty.0, len as libc::c_uint)) }
+    }
+
+    pub fn vector_type(&self, ll_elt_ty: Type, len: usize) -> Type {
+        unsafe { Type(LLVMVectorType(ll_elt_ty.0, len as libc::c_uint)) }
+    }
+
+    pub fn llvm_type_from_rust_type<T: 'static>(&self) -> Type {
+        if TypeId::of::<T>() == TypeId::of::<u8>() {
+            self.int8_type()
+        } else if TypeId::of::<T>() == TypeId::of::<u16>() {
+            self.int16_type()
+        } else if TypeId::of::<T>() == TypeId::of::<u32>() {
+            self.int32_type()
+        } else if TypeId::of::<T>() == TypeId::of::<u64>() {
+            self.int64_type()
+        } else if TypeId::of::<T>() == TypeId::of::<u128>() {
+            self.int128_type()
+        } else {
+            todo!("{}", std::any::type_name::<T>());
+        }
```

Comment:
This seems to work (but I don't insists it's better). Up to you if you want to change your variant.
```suggestion
        match type_name::<T>() {
         "u8" => self.int8_type(),
         "u16" => self.int16_type(),
         "u32" => self.int32_type(),
         "u64" => self.int64_type(),
         "u128" => self.int128_type(),
         _ => todo!("{}", std::any::type_name::<T>())
        }
```

Reply:
Thanks, this is more concise and I've changed it as you suggest.